app/controllers/teachers/students_controller.rb

```diff
@@ -1,15 +1,22 @@
 module Teachers
   class StudentsController < ApplicationController
     before_action :authenticate_user!, :set_user
+    before_action :set_student, only: [:show]
 
     def index
       @students = User.find(@user.group_members.map(&:user_id))
     end
 
+    def show; end
+
     private
 
+    def set_student
+      @student = User.find(params[:id])
+    end
+
     def set_user
-      user_id = current_user.id || params[:id]
+      user_id = current_user.id || params[:teacher_id]
```

Comment:
If user **will away** be authenticated...is it possible for `current_user.id` to return nil?